x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java

```diff
@@ -437,7 +438,8 @@ static RoleDescriptor kibanaSystem(String name) {
                         "index",
                         "delete",
                         TransportIndicesAliasesAction.NAME,
-                        TransportUpdateSettingsAction.TYPE.name()
+                        TransportUpdateSettingsAction.TYPE.name(),
+                        TransportAutoPutMappingAction.TYPE.name()
```

Comment:
@animehart Just want to clarify my understanding. The `auto_put` mapping privilege is being added because in v9.0, it will no longer be implicitly included/allowed with just the `index` privilege. And the `auto_put` mapping privilege is needed to be able to update index templates when new indices are created. Is that correct?

Reply:
Hi, per my understanding based on the discussion [here](https://elastic.slack.com/archives/C04F6U7QZ7B/p1732123917378789?thread_ts=1730930605.909099&cid=C04F6U7QZ7B), yes

tagging @maxcold so we all in the same page

Reply:
Yes, this is correct. However, we only experience the deprecation warning with `logs-cloud_security_posture.scores-default*`, so it might be a good idea to add `auto_put` only to this index pattern.

Reply:
> we only experience the deprecation warning with logs-cloud_security_posture.scores-default*, so it might be a good idea to add auto_put only to this index pattern
++ on this
If we can limit where the additional privilege is granted then we should.